doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,411 @@
+3. Add an option `-y/--year` that fills the variable `year` which will enable our program to filter the data by only including a season if it got released after the value `year`.
+4. Add an option `-a/--aggregate-by` that fills the variable `aggregate_by` which will enable our program choose between aggregating by mean or median.
+5. Add a flag `-H/--header-is-set` that fills the variable `notation`  which let the program know if it should ignore the first line in the file.
+
+Take a look at the help page again after you've done all of the above. You will notice that your options have been automatically included. **Copy and paste the example data file from the introduction** and check if your options are set correctly by trying the following few calls:
+
+```console
+./game_of_parsing -H -y 2014 data.tsv
+7.9175
+```
+
+```console
+./game_of_parsing -H -y 2010 --aggregate-by median data.tsv
+6.84
+```
+\endassignment
+\solution
+\include doc/tutorial/argument_parser/solution3.cpp
+\endsolution
+
+# List options
+
+In some use cases you may want to allow the user to specify an option multiple times and store the values in a list. With the seqan3::argument_parser this behaviour can be achieved simply by choosing your input variable to be of a container type (e.g. std::vector). The parser registers the container type through the seqan3::container_concept and will adapt the parsing of command line arguments accordingly.
+
+Example:
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp option_list
+
+Adding this option to a parser will allow you to call the program like this:
+
+```console
+./some_program -n Jon -n Arya -n Ned
+```
+
+The vector `list_variable` will then contain all three names `["Jon", "Arya", "Ned"]`.
+
+## Positional list options?
+
+An arbitrary positional option cannot be a list because of the ambiguity of which value belongs to which positional option. We do allow the very last option to be a list though for convenience. Note that if you try to add a positional list option which is not the last positional option, a seqan3::parser_design_error will be thrown.
+
+Example:
+
+\snippet doc/tutorial/argument_parser/small_snippets.cpp positional_option_list
+
+Adding these positional option to a parser will allow you to call the program like this:
+
+```console
+./some_program Stark Jon Arya Ned
+```
+
+The first `variable` will be filled with the value `Stark` while the vector `list_variable` will then contain all three names `["Jon", "Arya", "Ned"]`.
+
+\assignment{Assignment 4}
+
+Again, we extend the solution from assignment 3:
+
+1. Remove the option `-y/year`, since we want to keep it simple and only aggregate by season now.
+
+2. Add a variable `seasons` of type `std::vector<uint8_t>` to the struct `cmd_arguments`.
+
+3. Add a list option `-s/--season` that will fill the variable `seasons`, which let's the user specify which seasons to aggregate instead of the year.
```

Comment:
```suggestion
3. Add a list option `-s/--season` that will fill the variable `seasons` which lets the user specify which seasons to aggregate instead of the year.
```